io.welle.welle-gui.yml

```diff
@@ -0,0 +1,103 @@
+app-id: io.welle.welle-gui
+runtime: org.kde.Platform
+runtime-version: '6.8'
+sdk: org.kde.Sdk
+command: welle-io
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--filesystem=xdg-documents'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--socket=fallback-x11'
+  - '--device=dri'
+  - '--device=all'
+
+modules:
+  - name: fftw
+    buildsystem: autotools
+    config-opts:
+      - '--enable-shared'
+      - '--disable-static'
+      - '--enable-threads'
+      - '--enable-float'
+    sources:
+      - type: archive
+        url: 'http://fftw.org/fftw-3.3.10.tar.gz'
+        md5: 8ccbf6a5ea78a16dbc3e1306e234cc5c
+
+  - name: libfaad
+    buildsystem: cmake
+    sources:
+      - type: git
+        url: 'https://salsa.debian.org/multimedia-team/faad2.git'
+        tag: debian/2.11.2-1
+
+  - name: libusb
+    buildsystem: autotools
+    config-opts:
+      - '--disable-udev'
+    sources:
+      - type: git
+        url: 'https://github.com/libusb/libusb.git'
+        tag: v1.0.27
+
+  - name: librtlsdr
+    buildsystem: cmake
+    config-opts:
+      - '-Wno-dev'
+      - '-DDETACH_KERNEL_DRIVER=ON'
+    sources:
+      - type: git
+        url: 'https://gitea.osmocom.org/sdr/rtl-sdr.git'
+        tag: v2.0.2
+
+  - name: libairspy
+    buildsystem: cmake
+    config-opts:
+      - '-Wno-dev'
+    sources:
+      - type: git
+        url: 'https://github.com/airspy/airspyone_host.git'
+        tag: v1.0.10
+
+  - name: SoapySDR
+    buildsystem: cmake
+    config-opts:
+      - '-DCMAKE_BUILD_TYPE=Release'
+      - '-Wno-dev'
+    sources:
+      - type: git
+        url: 'https://github.com/pothosware/SoapySDR.git'
+        tag: soapy-sdr-0.8.1
+
+  - name: welle.io
+    buildsystem: qmake
+    build-commands:
+      # Copy welle.io binary
+      - cp src/welle-gui/welle-io /app/bin
+
+      # Copy icons
+      - install -Dm644 src/welle-gui/icons/16x16/welle-io.png /app/share/icons/hicolor/16x16/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/32x32/welle-io.png /app/share/icons/hicolor/32x32/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/48x48/welle-io.png /app/share/icons/hicolor/48x48/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/128x128/welle-io.png /app/share/icons/hicolor/128x128/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/256x256/welle-io.png /app/share/icons/hicolor/256x256/apps/io.welle.welle-gui.png
+      - install -Dm644 io.welle.welle-gui.desktop -t /app/share/applications
+
+      # Add current build date into appdata.xml
+      - sed -i 's/date="1970-01-01"/'date=\""$(date +%Y-%m-%d)"\"'/g' io.welle.welle-gui.appdata.xml
+      # Add app version and git hash into appdata.xml
+      - sed -i 's/version="0.0.0"/'version=\""$(cat src/welle-gui/_current_version)-$(git rev-parse --short HEAD)"\"'/g' io.welle.welle-gui.appdata.xml
```

Comment:
this should really be handled upstream as they are the one doing releases.

Reply:
also `sed` is terrible as it doesn't have any failure mode. use a patch instead.

Reply:
Do you have a succession how to put the version number and git hash into the manifest without doing a git commit? I would like to decrease the number of files that I have to change manually when doing a release.

Is your succession placing patches in this repo here?

Reply:
Fixed in https://github.com/AlbrechtL/flathub/commit/399692c6c6273b50da796fa854a789f567555523.